glib/src/enums.rs

```diff
@@ -551,75 +575,79 @@ impl FlagsClass {
+        }
     }
 }
 
 /// Representation of a single flags value of a `FlagsClass`.
-#[derive(Debug, Clone)]
 #[doc(alias = "GFlagsValue")]
-pub struct FlagsValue(*const gobject_ffi::GFlagsValue, FlagsClass);
+#[repr(transparent)]
+pub struct FlagsValue(gobject_ffi::GFlagsValue);
 
 unsafe impl Send for FlagsValue {}
 unsafe impl Sync for FlagsValue {}
 
+impl fmt::Debug for FlagsValue {
+    fn fmt(&self, f: &mut fmt::Formatter<'_>) -> fmt::Result {
+        f.debug_struct("FlagsValue")
+            .field("value", &self.value())
+            .field("name", &self.name())
+            .field("nick", &self.nick())
+            .finish()
+    }
+}
+
 impl FlagsValue {
     /// Get integer value corresponding to the value.
     #[doc(alias = "get_value")]
     pub fn value(&self) -> u32 {
-        unsafe { (*self.0).value }
+        self.0.value
     }
 
     /// Get name corresponding to the value.
     #[doc(alias = "get_name")]
     pub fn name(&self) -> &str {
-        unsafe { CStr::from_ptr((*self.0).value_name).to_str().unwrap() }
+        unsafe { CStr::from_ptr(self.0.value_name).to_str().unwrap() }
     }
 
     /// Get nick corresponding to the value.
     #[doc(alias = "get_nick")]
     pub fn nick(&self) -> &str {
-        unsafe { CStr::from_ptr((*self.0).value_nick).to_str().unwrap() }
+        unsafe { CStr::from_ptr(self.0.value_nick).to_str().unwrap() }
     }
 
     /// Convert flags value to a `Value`.
-    pub fn to_value(&self) -> Value {
+    pub fn to_value(&self, flags: &FlagsClass) -> Value {
         unsafe {
-            let mut v = Value::from_type(self.1.type_());
-            gobject_ffi::g_value_set_flags(v.to_glib_none_mut().0, (*self.0).value);
+            let mut v = Value::from_type(flags.type_());
+            gobject_ffi::g_value_set_flags(v.to_glib_none_mut().0, self.0.value);
             v
         }
     }
 
     /// Convert flags values from a `Value`. This returns all flags that are set.
-    pub fn from_value(value: &Value) -> Vec<FlagsValue> {
+    pub fn from_value(value: &Value) -> Option<(FlagsClass, Vec<&FlagsValue>)> {
         unsafe {
-            let flags_class = FlagsClass::new(value.type_());
+            let flags_class = FlagsClass::new(value.type_())?;
             let mut res = Vec::new();
-            if let Some(flags_class) = flags_class {
-                let f = gobject_ffi::g_value_get_flags(value.to_glib_none().0);
-                for v in flags_class.values() {
-                    if v.value() & f != 0 {
-                        res.push(v);
-                    }
+            let f = gobject_ffi::g_value_get_flags(value.to_glib_none().0);
+            for v in flags_class.values() {
+                if v.value() & f != 0 {
+                    res.push(&*(v as *const FlagsValue));
                 }
             }
-            res
+            Some((flags_class, res))
         }
     }
-
-    /// Get `FlagsClass` to which the flags value belongs.
```

Comment:
I suppose it's fine to lose access to the FlagClass/EnumClass

Reply:
Yes, you can't really have the value without the class.